docs/source/checks/tabular/model_evaluation/plot_weak_segments_performance.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""
+Weak Segments Performance
+********************************
+
+This notebooks provides an overview for using and understanding the weak segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+=================================
+The check is designed to help you easily identify weak spots of your model and provide a deepdive analysis into
+its performance on different segments of your data. Specifically, it is designed to help you identify the model weakest
+segments in the data distribution for further improvement and visibility purposes.
+
+In order to achieve this, the check trains several simple tree based models which try to predict the error of the
+user provided model on the dataset. The relevant segments are detected by analyzing the different
+leafs of the trained trees.
+"""
+#%%
+# Generate data & model
+# =====================
+
+from deepchecks.tabular.datasets.classification.phishing import (
+    load_data, load_fitted_model)
+
+train_dataset, test_dataset = load_data()
+model = load_fitted_model()
+
+#%%
+# Run the check
+# =============
+# The check have several key parameters that can define the behavior of the check and especially its output.
+#
+# alternative_scorer: Determines the metric to be used as the performance measurement of the model on different
+# segments. It is important to select a metric that is relevant to the data domain and task you are performing.
+# By default, the check uses Neg RMSE for regression tasks and Accuracy for classification tasks.
+#
+# segment_minimum_size_ratio: Determines the minimum size of segments that are of interest. The check is tuned to find
+# the weakest segment regardless of the segment size and so it is recommended to try different configurations of this
+# parameter as larger segments can be of interest even the model performance on them is superior.
+#
+# columns / ignore_columns: Controls which columns should be searched for weak segments. By default,
+# a heuristic is used to determine which columns to use based on their feature importance.
+#
+# categorical_aggregation_threshold: By default the check will combine rare categories into a single category called
+# "other". This parameter determines the frequency threshold for categories to be mapped into to the "other" category.
```

Comment:
1. Link to API reference at the beginning to see all params (note - see 4 underneath). see how this is done in the sources of the train_test_validation quickstart: [Link to Dataset Object](https://docs.deepchecks.com/stable/user-guide/tabular/auto_tutorials/plot_quick_train_test_validation.html#define-a-dataset-object), or [link to lending_club](https://docs.deepchecks.com/stable/user-guide/tabular/auto_tutorials/plot_quick_train_test_validation.html#quickstart-train-test-validation-suite-loans-data)
2. Would put here the `columns` / `ignore_columns` first + and/or emphasize it more in the explanation. Also in demonstration would add that parameter...
3. Recommended - all parameters explained here in the Run Check demonstration so that it's easy to copy_paste + if rename happens it will fail and we'll notice it...
4. add to all params either - link to api reference (this is best and then 1 above can be dropped) - have to check how to do it with rst but this is possible (since the sphinx builds it all. It's possible to link to modules, classes, functions, etc. I don't know if possible to link to params). If doesn't work out then at least the `` for rst parsing - so it is parsed visually like code parameters (in pink)

Reply:
Regarding 4, can you rephrase? I did not understand / found what you meant for in the train_test_validation quickstart